sound/soc/sof/ipc4.c

```diff
@@ -516,7 +513,8 @@ static int sof_ipc4_set_get_data(struct snd_sof_dev *sdev, void *data,
 	if (!set && payload_bytes != offset)
 		ipc4_msg->data_size = offset;
 
-	sof_ipc4_dump_payload(sdev, ipc4_msg->data_ptr, ipc4_msg->data_size);
+	if (sof_debug_check_flag(SOF_DBG_DUMP_IPC_MESSAGE_PAYLOAD))
+		sof_ipc4_dump_payload(sdev, ipc4_msg->data_ptr, ipc4_msg->data_size);
```

Comment:
if we introduce a flag, don't we want to support IPC3 as well? I am not clear as to why this is only for IPC4.

Reply:
@plbossart, we never had this done for IPC3, I have looked at it briefly, but it is a bit more involved it looks like but I'm planning to have a separate PR to rework the logging to make space for the whole message dumping.
We, for example are not printing 'payload' size as it makes no sense for IPC3 - the whole message, including header is in the payload area.
The way IPC3 works is substantially different in this level compared to IPC4, but I have few ideas to improve it.

It is going to be a separate PR.